sycl/include/sycl/ext/oneapi/experimental/graph.hpp

```diff
@@ -509,14 +515,91 @@ class __SYCL_EXPORT dynamic_parameter_base {
   void updateValue(const raw_kernel_arg *NewRawValue, size_t Size);
 
   void updateAccessor(const sycl::detail::AccessorBaseHost *Acc);
+
+  void updateWorkGroupMem(size_t BufferSize);
+
   std::shared_ptr<dynamic_parameter_impl> impl;
 
   template <class Obj>
   friend const decltype(Obj::impl) &
   sycl::detail::getSyclObjImpl(const Obj &SyclObject);
 };
+
+class dynamic_work_group_memory_base : public dynamic_parameter_base {
+public:
+  dynamic_work_group_memory_base(
+      experimental::command_graph<graph_state::modifiable> Graph, size_t Size)
+      : dynamic_parameter_base(Graph), BufferSize(Size) {}
+
+private:
+  size_t BufferSize{};
+  friend class sycl::handler;
+};
 } // namespace detail
 
+template <typename DataT, typename PropertyListT = empty_properties_t>
+class __SYCL_SPECIAL_CLASS
+__SYCL_TYPE(dynamic_work_group_memory) dynamic_work_group_memory
```

Comment:
Does `dynamic_work_group_memory` need common reference semantics defined for it? for example https://github.com/intel/llvm/pull/16788

Reply:
As discussed offline, the `dynamic_parameter_base` implements `==` and `!=` operators which should be satisfactory for this derived class too. I've added the `std::hash` though.